### PR TITLE
Add ChefSpec Unit Tests

### PR DESCRIPTION
**Pull request by pwelch**
Add ChefSpec and resources for unit test coverage of current cookbook resources.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,216 @@
+GEM
```
Comment (tas50) [inline Gemfile.lock]: Gemfile.lock shouldn't get added to the cookbook

Comment (pwelch) [inline Gemfile.lock]: - Removed Gemfile.lock
- Added Gemfile.lock to .gitignore
- Rebased and pushed updated branch. You may need to fetch to get the latest branch.

**Comment by tas50** [comment]
@pwelch If you wanna rebase this I'd love to get it merged in

**Comment by pwelch** [comment]
@tas50 Rebased and fixed some Rubocop warnings. Let me know if you need anything from me.

Thanks!

**Pull request merged**